src/app/connections/views/ConnectionOutboundSummaryOverviewView.vue

```diff
@@ -321,12 +321,20 @@ import PolicyTypeTag from '@/app/common/PolicyTypeTag.vue'
 import type { DataplaneOverview } from '@/app/data-planes/data/'
 import type { PolicyTypeCollectionSource } from '@/app/policies/sources'
 import RuleMatchers from '@/app/rules/components/RuleMatchers.vue'
+import { ResourceRule } from '@/app/rules/data/ResourceRule'
 import { sources } from '@/app/rules/sources'
 
 const props = defineProps<{
   data: Record<string, any>
   dataplaneOverview: DataplaneOverview
 }>()
+
+const ruleForCluster = (cluster: any, rule: ResourceRule) => {
+  return cluster.$resourceMeta.name === rule.resourceMeta.labels['kuma.io/display-name'] &&
+  cluster.$resourceMeta.namespace === rule.resourceMeta.labels['k8s.kuma.io/namespace'] &&
+  cluster.$resourceMeta.zone === rule.resourceMeta.labels['kuma.io/zone'] &&
+  (rule.resourceSectionName === '' || cluster.$resourceMeta.port === rule.resourceSectionName)
+}
```

Comment:
The main change is here:

We use the labels in the resourceMeta, instead of the other things in the resourceMeta to lookup the connection between a clusterName and a rule.

Additionally, if resourceMeta.resourceSectionName is non-empty, we also check that it equals the port form the cluster name.